app/src/main/java/com/mapbox/navigation/examples/junctions/ShowJunctionsActivity.kt

```diff
@@ -0,0 +1,295 @@
+import com.mapbox.navigation.ui.maps.route.line.api.MapboxRouteLineView
+import com.mapbox.navigation.ui.maps.route.line.model.MapboxRouteLineOptions
+import com.mapbox.navigation.ui.maps.route.line.model.RouteLineResources
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+/**
+ * This activity demonstrates the usage of the [MapboxJunctionApi]. There is boiler plate
+ * code for establishing basic navigation and a route simulator is used. The example assumes
+ * that LOCATION permission has already been granted.
+ *
+ * The code specifically related to the junction component is commented in order to call
+ * attention to its usage. The example uses a predefined location pair to demonstrate junction.
+ * Click "Set Route" button to use predefined coordinates and trigger navigation.
+ *
+ * Note: A special access token is required to get access to junctions in directions response.
+ */
+class ShowJunctionsActivity : AppCompatActivity() {
+
+    private lateinit var mapboxMap: MapboxMap
+    private lateinit var mapboxNavigation: MapboxNavigation
+    private lateinit var binding: MapboxActivityShowJunctionsBinding
+    private lateinit var locationComponent: LocationComponentPlugin
+
+    private val mapboxReplayer = MapboxReplayer()
+    private val navigationLocationProvider = NavigationLocationProvider()
+
+    /**
+     * The [MapboxJunctionApi] consumes banner instructions data and produces junctions related
+     * data that is consumed by the [MapboxJunctionView] in the view layout.
+     *
+     * Uses a specific access token required for the route request to send junctions in the response.
+     */
+    private val junctionApi: MapboxJunctionApi by lazy {
+        MapboxJunctionApi(getString(R.string.mapbox_access_token))
+    }
+
+    /**
+     * The result of invoking [MapboxJunctionApi.generateJunction] is returned as a callback
+     * containing either a success in the form of [JunctionValue] or failure in the form of
+     * [JunctionError].
+     */
+    private val junctionCallback =
+        MapboxNavigationConsumer<Expected<JunctionError, JunctionValue>> { value ->
+            // The data obtained must be rendered by [MapboxJunctionView]
+            binding.junctionView.render(value)
+        }
+
+    private val routeLineResources: RouteLineResources by lazy {
+        RouteLineResources.Builder().build()
+    }
+
+    private val options: MapboxRouteLineOptions by lazy {
+        MapboxRouteLineOptions.Builder(this)
+            .withRouteLineResources(routeLineResources)
+            .withRouteLineBelowLayerId("road-label")
+            .build()
+    }
+
+    private val routeLineView by lazy {
+        MapboxRouteLineView(options)
+    }
+
+    private val routeLineApi: MapboxRouteLineApi by lazy {
+        MapboxRouteLineApi(options)
+    }
+
+    private val routeArrowApi: MapboxRouteArrowApi by lazy {
```

Comment:
We can remove codes related to `routeArrow` as it is not relevant to this example.